xapian-core/bin/xapian-replicate-server.cc

```diff
@@ -73,7 +73,12 @@ main(int argc, char **argv)
 		host.assign(optarg);
 		break;
 	    case 'p':
-		port = atoi(optarg);
+		if (!parse_signed(optarg, port) ||
+		   (port <= 0 || port >= 65536)) {
```

Comment:
When wrapping an expression at an operator, align the two operands vertically (so the `(` here should align under the `!` on the line above).

And a minor cosmetic point, but writing the range check as `port < 1 || port > 65535` would mean the values here match those in the error message, which seems slightly cleaner.

Reply:
(Ah, I see that range check was actually copied from elsewhere)